### PR TITLE
feat: Gsoc & tsoc page

### PR DESCRIPTION
**Pull request by ryuk156**
This PR introduces Gsoc & Tsoc page, which includes two-component  ProjectModal and Gsoc_Tsoc .
This page uses three npm packages to create Carousel and parse HTML and markdown format getting from API

1. https://www.npmjs.com/package/react-multi-carousel
2. https://www.npmjs.com/package/html-react-parser
3. https://www.npmjs.com/package/remarkable

Note: Navbar structure will be improved in separate PR

### CODE REVIEW

File: Module-generation/frontmatter.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import os
```
Comment (jdrueckert) [inline Module-generation/frontmatter.py]: Why is this file in here?
Looks like you did not create this branch from master but from another feature branch :thinking:

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  useEffect(() => {
+    readyIdeas();
+    ongoingIdeas();
+  }, []);
+
+  return (
+    <div>
+      <div>
+        <h4 className="text-center">
+          About Google summer of code & Terasology summer of code{" "}
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx:76]:
```suggestion
          GSoC & TSoC @ Terasology{" "}
```

File: src/components/ProjectModal/ProjectModal.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  return (
+    <div>
+      <div className="mt-4 ">
+        <button
+          type="button"
+          onClick={toggle}
+          className="btn btn-lg btn-primary border border-0 p-2 font-weight-bold"
+        >
+          More about project
```
Comment (jdrueckert) [inline src/components/ProjectModal/ProjectModal.jsx]:
```suggestion
          Project Details
```

File: src/components/ProjectModal/ProjectModal.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        </button>
+      </div>
+      <Modal className="modal-dialog modal-xl" isOpen={modal} toggle={toggle}>
+        <h4 className="mt-3 ml-3">{name}</h4>
+        <hr />
+        <div className="m-4">
+          <div className="m-3">{parse(descs)}</div>
+          <h6 className="mt-4 ml-2">Project tags:</h6>
```
Comment (jdrueckert) [inline src/components/ProjectModal/ProjectModal.jsx]:
```suggestion
          <h6 className="mt-4 ml-2">Project Tags:</h6>
```

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        </h4>
+        <div className="container my-4">
+          <div className="home-underline"></div>
+
+          <Row className="justify-content-center">
+            <div className="col-md-10">
+              <div className="text-center  gsoc_tsoc_content">
+                Google Summer of Code (GSoC) is a global program focused on
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx]:
```suggestion
                <b>Google Summer of Code (GSoC)</b> is a global program focused on
```

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                <a
+                  className="text-success font-weight-bold"
+                  href="https://summerofcode.withgoogle.com/"
+                >
+                  here
+                </a>
+              </div>
+
+              <div className="text-center mt-4 gsoc_tsoc_content">
+                Terasology Summer of Code (TSoC) is similar as Google Summer of
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx:93]:
```suggestion
                <b>Terasology Summer of Code (TSoC)</b> is similar as Google Summer of
```

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            <div className="col-md-10">
+              <div className="text-center  gsoc_tsoc_content">
+                Google Summer of Code (GSoC) is a global program focused on
+                bringing more student developers into open source software
+                development.Every year more than 1200 students are
+                accepted.Accepted students work with a mentor and become a part
+                of the open source community. Many become lifetime open source
+                developers.Spend your summer break writing code and learning
+                about open source development while earning a stipend. for more
+                you can visit{" "}
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx]:
```suggestion
                Google Summer of Code (GSoC) is a global program focused on
                bringing more student developers into open source software
                development. Every year more than 1200 students are
                accepted. Accepted students work with a mentor and become a part
                of the open source community. Many become lifetime open source
                developers. Spend your summer break writing code and learning
                about open source development while earning a stipend. For more
                information, visit{" "}
```

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+                  here
+                </a>
+              </div>
+
+              <div className="text-center mt-4 gsoc_tsoc_content">
+                Terasology Summer of Code (TSoC) is similar as Google Summer of
+                Code (GSoC).It is sponsored by Teraslogy with different Timeline
+                then GSoC, do half of work, take a break for exams and other
+                thing and then do second half work .Lower stipend then GSoC. for
+                more information join our{" "}
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx]:
```suggestion
                Terasology Summer of Code (TSoC) is similar to Google Summer of 
                Code (GSoC). It provides more flexible conditions regarding time and 
                project scope than GSoC to also suit students that do not have a 
                10-week break over the summer. However, as it is sponsored directly 
                by Terasology the stipends are lower than with GSoC. For 
                more information join our{" "}
```

**Review by jdrueckert — CHANGES_REQUESTED** [review]
Minor suggestions for text-wise improvements.
@majordwarf, @pollend please have a code-wise look.

File: Module-generation/frontmatter.py (hunk 1 above)
Comment (ryuk156) [reply Module-generation/frontmatter.py]: oh , I stash changes and did pop , will remove this files

**Comment by ryuk156** [comment]
Done with above changes

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+            <div className="col-md-10">
+              <div className="text-center  gsoc_tsoc_content">
+                <b> Google Summer of Code (GSoC) </b> is a global program
+                focused on bringing more student developers into open source
+                software development.Every year more than 1200 students are
+                accepted.Accepted students work with a mentor and become a part
+                of the open source community. Many become lifetime open source
+                developers.Spend your summer break writing code and learning
+                about open source development while earning a stipend. for more
+                information visit{" "}
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx]: You missed something :sweat_smile: 
```suggestion
                <b> Google Summer of Code (GSoC) </b> is a global program
                focused on bringing more student developers into open source
                software development. Every year more than 1200 students are
                accepted. Accepted students work with a mentor and become a part
                of the open source community. Many become lifetime open source
                developers. Spend your summer break writing code and learning
                about open source development while earning a stipend. for more
                information visit{" "}
```

Comment (ryuk156) [reply src/components/GsocTsoc/GsocTsoc.jsx]: done also added tags on cards please have a look

**Comment by majordwarf** [comment]
Screenshot 2021-07-08 at 3 47 48 PM
 Why does the modal leave so much space on top of the page? Was this something you intended?

**Comment by ryuk156** [comment]
> Screenshot 2021-07-08 at 3 47 48 PM
> 
> Why does the modal leave so much space on top of the page? Was this something you intended?

done, please have a look again

**Comment by jdrueckert** [comment]
This has conflicts that need to be resolved

**Comment by ryuk156** [comment]
> This has conflicts that need to be resolved

Removed please have a look

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+            <div className="col-md-10">
+              <div className="text-center  gsoc_tsoc_content">
+                <b> Google Summer of Code (GSoC) </b> is a global program
+                focused on bringing more student developers into open source
+                software development. Every year more than 1200 students are
+                accepted. Accepted students work with a mentor and become a part
+                of the open source community. Many become lifetime open source
+                developers. Spend your summer break writing code and learning
+                about open source development while earning a stipend. for more
+                information visit{" "}
```
Comment (jdrueckert) [inline src/components/GsocTsoc/GsocTsoc.jsx]:
```suggestion
                <b>Google Summer of Code (GSoC)</b> is a global program
                focused on bringing more student developers into open source
                software development. Every year more than 1200 students are
                accepted. Accepted students work with a mentor and become a part
                of the open source community. Many become lifetime open source
                developers. Spend your summer break writing code and learning
                about open source development while earning a stipend. For more
                information visit{" "}
```

Comment (ryuk156) [reply src/components/GsocTsoc/GsocTsoc.jsx]: done

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 11]
```diff
@@ -0,0 +1,239 @@
+import React, { useState, useEffect } from "react";
+import { Table, NavLink, Input, Badge, Row, Col, Alert } from "reactstrap";
+import Carousel from "react-multi-carousel";
+import "react-multi-carousel/lib/styles.css";
+import ProjectModal from "../ProjectModal/ProjectModal";
+import defaultcardimg from "../../../static/logos/defaultCardcover.jpg";
+
+const GsocTsoc = () => {
```
Comment (majordwarf) [inline src/components/GsocTsoc/GsocTsoc.jsx:8]: Be consistent with variable naming conventions, we're using camel-case variables everywhere else.

File: src/components/ProjectModal/ProjectModal.jsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+import { Badge, Modal, ModalFooter } from "reactstrap";
+import { Remarkable } from "remarkable";
+import parse from "html-react-parser";
+
+const ProjectModal = ({ name, desc, tags }) => {
+  const [modal, setModal] = useState(false);
+  const toggle = () => setModal(!modal);
+  const md = new Remarkable();
+
+  const descs = md.render(desc);
```
Comment (majordwarf) [inline src/components/ProjectModal/ProjectModal.jsx]: Can we name it something more appropriately? `description`?

**Review by majordwarf — CHANGES_REQUESTED** [review]

File: src/components/GsocTsoc/GsocTsoc.jsx [hunk 13]
```diff
@@ -1,9 +1,9 @@
 import React, { useState, useEffect } from "react";
-import { Table, NavLink, Input, Badge, Row, Col, Alert } from "reactstrap";
+import { Row,Alert } from "reactstrap";
```
Comment (majordwarf) [inline src/components/GsocTsoc/GsocTsoc.jsx]: Space after `,`

**Review by majordwarf — CHANGES_REQUESTED** [review]
When you open the project card, you can see a horizontal scrollbar. This is a CSS fix.

File: src/components/GsocTsoc/GsocTsoc.jsx (hunk 13 above)
Comment (ryuk156) [reply src/components/GsocTsoc/GsocTsoc.jsx]: updated

**Review by majordwarf — APPROVED** [review]

**Pull request merged**